Include an auth header if FPL_REGISTRY_TOKEN is present

bkaney opened the pull request:
**Description:**
Supports an optional `FPL_REGISTRY_TOKEN` ENV to set an authentication header where a package registry needs to  include auth credentials

bkaney committed:
Include an auth header if FPL_TOKEN is present

bkaney committed:
Restore old ENV after tests

bkaney committed:
Update README

cmoesel reviewed (commented):
Thanks for this, @bkaney. This looks good (and is very straight-forward, which is always nice). I have a few questions:

1. Have you tested this with your registry to confirm it really works in the real world (outside of unit tests)?
2. How would you feel about `FPL_REGISTRY_TOKEN` instead of `FPL_TOKEN`? To me, that more clearly couples it to the `FPL_REGISTRY` option -- and it reduces the opportunity for confusion in the future if FPL ever needs to support tokens for other sources.

bkaney committed:
Change to FPL_REGISTRY_TOKEN

bkaney committed:
Provide error hint when unauthorized

401s errors were getting swallowed in the error handling, and for end-user experience it is nice to know when there is an auth problem.

bkaney commented:
@cmoesel  - We did a bunch of testing. I changed to FPL_REGISTRY_TOKEN. And also added some handling so users know if they are running into auth errors. We ran into this, and really thought it was important to have that message to diagnose problems. Thanks!

cmoesel reviewed (approved):
This looks good! Thanks for the contribution, @bkaney!
